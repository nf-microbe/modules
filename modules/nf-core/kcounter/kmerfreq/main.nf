process KCOUNTER_KMERFREQ {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/kcounter_biopython_numpy:ec6e495e10ab3f87':
        'community.wave.seqera.io/library/kcounter_biopython_numpy:2944df66b2e402a3' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("${prefix}.kcounter_kmerfreq.tsv")    , emit: tsv
    path "versions.yml"                                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    kmer_freq.py \\
        --input ${fasta} \\
        --output ${prefix}.kcounter_kmerfreq.tsv \\
        --threads ${task.cpus} \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
        python: \$( python --version | sed 's/Python //' )
        kcounter: 0.1.1
        numpy: \$(python -c "import numpy; print(numpy.__version__)")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.kcounter_kmerfreq.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
        python: \$( python --version | sed 's/Python //' )
        kcounter: 0.1.1
        numpy: \$(python -c "import numpy; print(numpy.__version__)")
    END_VERSIONS
    """
}
