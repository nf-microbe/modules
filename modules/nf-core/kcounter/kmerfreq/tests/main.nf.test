nextflow_process {

    name "Test Process KCOUNTER_KMERFREQ"
    script "../main.nf"
    process "KCOUNTER_KMERFREQ"

    tag "modules"
    tag "modules_nfcore"
    tag "kcounter"
    tag "kcounter/kmerfreq"

    test("bacteroides_fragilis - illumina - fasta") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("bacteroides_fragilis - genome - fasta - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/genome/genome.fna.gz', checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}