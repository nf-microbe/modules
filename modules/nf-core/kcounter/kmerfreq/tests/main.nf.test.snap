{
    "bacteroides_fragilis - genome - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.kcounter_kmerfreq.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,96d0faf51ae4246803f4787ad85fb595"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.kcounter_kmerfreq.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,96d0faf51ae4246803f4787ad85fb595"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T07:32:37.535550501"
    },
    "bacteroides_fragilis - illumina - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.kcounter_kmerfreq.tsv:md5,90347d7a7b3b78788c03fea3d6586067"
                    ]
                ],
                "1": [
                    "versions.yml:md5,96d0faf51ae4246803f4787ad85fb595"
                ],
                "tsv": [
                    [
                        {
                            "id": "test"
                        },
                        "test.kcounter_kmerfreq.tsv:md5,90347d7a7b3b78788c03fea3d6586067"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,96d0faf51ae4246803f4787ad85fb595"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-15T07:32:22.764295386"
    }
}